src/main/java/com/aws/greengrass/componentmanager/ComponentServiceHelper.java

```diff
@@ -84,14 +85,32 @@ ResolvedComponentVersion resolveComponentVersion(String componentName, Semver lo
                         .maxRetryInterval(retryInterval).maxAttempt(CLIENT_RETRY_COUNT)
                         .retryableExceptions(Arrays.asList(DeviceConfigurationException.class)).build();
 
-        try (GreengrassV2DataClient greengrasV2DataClient = RetryUtils.runWithRetry(clientExceptionRetryConfig,
+        try (GreengrassV2DataClient greengrassV2DataClient = RetryUtils.runWithRetry(clientExceptionRetryConfig,
                 clientFactory::fetchGreengrassV2DataClient, "get-greengrass-v2-data-client", logger)) {
-            result = greengrasV2DataClient.resolveComponentCandidates(request);
+            result = greengrassV2DataClient.resolveComponentCandidates(request);
         } catch (ResourceNotFoundException e) {
-            logger.atDebug().kv("componentName", componentName).kv("versionRequirements", versionRequirements)
-                    .log("No applicable version found in cloud registry", e);
-            throw new NoAvailableComponentVersionException("No cloud component version satisfies the requirements.",
-                    componentName, versionRequirements);
+            if (e.getMessage() == null) {
+                throw new PackagingException(
+                    "Component service returns invalid exception message",
+                    DeploymentErrorCode.RESOLVE_COMPONENT_CANDIDATES_BAD_RESPONSE);
```

Comment:
This is a cloud service error, which we count as a system error. I don't think we have any commitment from cloud to never return a null message, and in this case we know that the exception type is ResourceNotFoundException.

I would recommend to just throw a NoAvailableComponentVersionException with default message if the exception message is null.